clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -276,22 +284,24 @@ public void poll(long timeout) throws IOException {
         long startSelect = time.nanoseconds();
         int readyKeys = select(timeout);
         long endSelect = time.nanoseconds();
-        currentTimeNanos = endSelect;
         this.sensors.selectTime.record(endSelect - startSelect, time.milliseconds());
 
         if (readyKeys > 0 || !immediatelyConnectedKeys.isEmpty()) {
-            pollSelectionKeys(this.nioSelector.selectedKeys(), false);
-            pollSelectionKeys(immediatelyConnectedKeys, true);
+            pollSelectionKeys(this.nioSelector.selectedKeys(), false, endSelect);
+            pollSelectionKeys(immediatelyConnectedKeys, true, endSelect);
         }
 
         addToCompletedReceives();
 
         long endIo = time.nanoseconds();
         this.sensors.ioTime.record(endIo - endSelect, time.milliseconds());
-        maybeCloseOldestConnection();
+
+        maybeCloseOldestConnection(endIo);
```

Comment:
This is a change in semantics, right? Before, we would never expire a connection that has been processed during the `poll` because we use `currentTimeNanos`. After this change, if something in `poll` takes long enough, we could end up expiring a connection in `completedReceived` for example.

Reply:
this problem also affect replication node fetch process? our server occur a error:

Reply:
this problem also affect replication node fetch process? our server occur a error:
[2019-04-05 23:59:46,084] WARN [ReplicaFetcherThread-1-1], Error in fetch kafka.server.ReplicaFetcherThread$FetchRequest@b22a64c (kafka.server.ReplicaFetcherThread)
java.io.IOException: Connection to 1 was disconnected before the response was read
but return to normal after a reboot kafka server.